### PR TITLE
Recorded future event collector

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5028

## Description
Added a new event collector for RecordedFututre to XSIAM.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+This is the Recorded Future event collector integration for XSIAM.
+This integration was integrated and tested with version 2 of Recorded Future API
```
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md:2]:
```suggestion
This integration was integrated and tested with version 2 of the Recorded Future API.
```

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Configure Recorded Future Event Collector on Cortex XSOAR
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
+2. Search for Recorded Future Event Collector.
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | API token | The API token to use for connection | True |
```
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md:12]:
```suggestion
    | API token | The API token to use for the connection. | True |
```

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/RecordedFutureEventCollector.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+category: Analytics & SIEM
+sectionOrder:
+- Connect
+- Collect
+commonfields:
+  id: RecordedFutureEventCollector
+  version: -1
+configuration:
+- displaypassword: API token
+  additionalinfo: The API token to use for connection
```
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/RecordedFutureEventCollector.yml:10]:
```suggestion
  additionalinfo: The API token to use for the connection.
```

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    | Trust any certificate (not secure) | Use SSL secure connection or not. | False |
+    | Use system proxy settings | Use proxy settings for connection or not. | False |
+    | First fetch time |  First fetch query `<number> <time unit>`, e.g., `7 days`. Default `3 days` | False |
+    | Max fetch | The maximum number of events per fetch, Default and maximum is 1000. | False |
```
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md:16]:
```suggestion
    | Max fetch | The maximum number of events per fetch. Default and maximum is 1000. | False |
```

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/RecordedFutureEventCollector.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  type: 8
+  section: Connect
+- defaultvalue: 3 days
+  display: First fetch time
+  name: first_fetch
+  required: false
+  type: 0
+  section: Collect
+- defaultvalue: 1000
+  display: The maximum number of events per fetch, Default and maximum is 1000.
```
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/RecordedFutureEventCollector.yml:33]:
```suggestion
  display: The maximum number of events per fetch. Default and maximum is 1000.
```

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/RecordedFutureEventCollector.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      defaultValue: 'false'
+      description: If true, the command will create events, otherwise it will only display them.
+      isArray: false
+      name: should_push_events
+      predefined:
+      - 'true'
+      - 'false'
+      required: true
+    - defaultValue: 10
+      description: Maximum results to return.
```
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/RecordedFutureEventCollector.yml:54]:
```suggestion
      description: Maximum number of results to return.
```

File: Packs/RecordedFuture/ReleaseNotes/1_3_4.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+##### New: Recorded Future Event Collector
+This is the Recorded Future event collector integration for XSIAM. (Available from Cortex XSIAM 1.3).
```
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/ReleaseNotes/1_3_4.md:4]:
```suggestion
This is the Recorded Future Event Collector integration for XSIAM. (Available from Cortex XSIAM 1.3).
```

**Comment by ShirleyDenkberg** [comment]
@gal-forer @DeanArbel Doc review completed.

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+This is the Recorded Future event collector integration for XSIAM.
+This integration was integrated and tested with version 2 of the Recorded Future API.
+
+## Configure Recorded Future Event Collector on Cortex XSOAR
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
```
Comment (gal-forer) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md:6]: Is this the right path  for XSIAM ?

File: Packs/RecordedFuture/pack_metadata.json [hunk 9]
```diff
@@ -2,7 +2,7 @@
     "name": "Recorded Future Intelligence",
     "description": "Recorded Future App, this pack is previously known as 'RecordedFuture v2'",
     "support": "partner",
-    "currentVersion": "1.3.3",
+    "currentVersion": "1.3.4",
```
Comment (gal-forer) [inline Packs/RecordedFuture/pack_metadata.json:5]: I think that adding a new collector to the pack should be more than a revision, have you asked the tpm/pm/ Dean ?

File: Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md (hunk 8 above)
Comment (ShirleyDenkberg) [inline Packs/RecordedFuture/Integrations/RecordedFutureEventCollector/README.md:6]:
```suggestion
1. Navigate to **Settings** > **Configurations** > **Data Collection** > **Automations & Feed Integrations**.
```

File: Packs/RecordedFuture/pack_metadata.json (hunk 9 above)
Comment (MosheEichler) [reply Packs/RecordedFuture/pack_metadata.json:5]: Regarding the new/existing pack question.
There is no way to add an integration with XSOAR support to a pack with partner support
I am waiting for a reply from @DanilVil.
So let's wait for Danil's reply (maybe we should move the integration to a different pack).

**Review by gal-forer — APPROVED** [review]

**Pull request merged**